src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -763,44 +763,52 @@ public SocketError ReceiveFrom(byte[] buffer, int offset, int count, ref SocketF
         {
             Debug.Assert(timeout == -1 || timeout > 0, $"Unexpected timeout: {timeout}");
 
-            SocketFlags receivedFlags;
-            SocketError errorCode;
-            if (SocketPal.TryCompleteReceiveFrom(_socket, buffer, offset, count, flags, socketAddress, ref socketAddressLen, out bytesReceived, out receivedFlags, out errorCode))
-            {
-                flags = receivedFlags;
-                return errorCode;
-            }
-
-            using (var @event = new ManualResetEventSlim(false, 0))
+            ManualResetEventSlim @event = null;
```

Comment:
I'll leave this for now; it's not much extra code, and it avoids an implicit dependency on an MRES implementation detail.